bee-common/bee-common/src/shutdown_stream.rs

```diff
@@ -52,18 +52,19 @@ impl<S: Stream> ShutdownStream<stream::Fuse<S>> {
             stream: stream.fuse(),
         }
     }
+
+    /// Consume and split the `ShutdownStream` into its shutdown receiver and stream.
+    pub fn split(self) -> (future::Fuse<oneshot::Receiver<()>>, futures::stream::Fuse<S>) {
```

Comment:
Perhaps it would be a good idea to hide these definitions behind a typedef?

Reply:
Will do